src/snowflake/cli/api/project/schemas/template.py

```diff
@@ -0,0 +1,66 @@
+# Copyright (c) 2024 Snowflake Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+from __future__ import annotations
+
+from enum import Enum
+from typing import Any, List, Optional
+
+from pydantic import BaseModel, Field
+from snowflake.cli.api.secure_path import SecurePath
+
+
+class TemplateVariableType(Enum):
+    STRING = "string"
+    INTEGER = "int"
+    FLOAT = "float"
```

Comment:
```suggestion
    STRING = str
    INTEGER = int
    FLOAT = float
```
Won't it work?

Reply:
Let's also add support for booleans. For booleans we should support [y/N] prompts by using `typer.confirm` instead of `typer.prompt`

Reply:
I thought about that, but they create problems with rendering. By default jinja renders True/False, but this works only for Python files - for example json will have syntax errors
WDYT?

Reply:
> Won't it work?
Unfortunately :(

```
3 validation errors for Template
variables.1.type
  Input should be <class 'str'>, <class 'int'> or <class 'float'> [type=enum, input_value='int', input_type=str]
    For further information visit https://errors.pydantic.dev/2.7/v/enum
```

Reply:
Thanks!